src/main/java/nova/core/util/EnumSelector.java

```diff
@@ -0,0 +1,236 @@
+	 * <p>
+	 * Use {@link #apart(java.lang.Enum)} to specify what should be allowed.
+	 *
+	 * @see #allowAll()
+	 * @see #apart(java.lang.Enum)
+	 * @return this
+	 * @throws IllegalStateException If the EnumSelector has been {@link #lock() locked}.
+	 */
+	public EnumSelector<T> blockAll() {
+		checkWritable();
+		if (!defaultAllow)
+			defaultBlock = true;
+		else
+			throw new IllegalStateException("You can't block all enum values when you are already allowing them.");
+		return this;
+	}
+
+	/**
+	 * Specify what {@code enum} values should have behavior opposite of the default
+	 *
+	 * @see #allowAll()
+	 * @see #blockAll()
+	 * @param value The given {@code enum} value that should have behavior opposite of the default.
+	 * @return this
+	 */
+	public EnumSelector<T> apart(T value) {
+		checkWritable();
+		exceptions.add(value);
+		return this;
+	}
+
+	/**
+	 * Lock the EnumSelector, making it immutable.
+	 * Required for using of all getter methods.
+	 *
+	 * @see #allowAll()
+	 * @see #blockAll()
+	 * @see #locked()
+	 * @return this
+	 * @throws IllegalStateException If the EnumSelector does not have specified default behavior.
+	 */
+	public EnumSelector<T> lock() {
+		if (defaultAllow || defaultBlock)
+			locked = true;
+		else
+			throw new IllegalStateException("Cannot lock EnumSelector without specifying default behaviour.");
+		return this;
+	}
+
+	/**
+	 * Check if the EnumSelector instance has been locked.
+	 *
+	 * @return The locked status.
+	 */
+	public boolean locked() {
+		return locked;
+	}
+
+	/**
+	 * Check if the {@code enum} value is allowed by this EnumSelector.
+	 *
+	 * @param value The {@code enum} value to test.
+	 * @return If the {@code enum} value is allowed by this EnumSelector.
+	 * @throws IllegalStateException If the EnumSelector has not been {@link #lock() locked}.
+	 */
+	public boolean allows(T value) {
+		checkReadable();
+		return defaultAllow ^ exceptions.contains(value);
+	}
+
+	/**
+	 * Check if the EnumSelector allows all values.
+	 *
+	 * @return If the EnumSelector allows all values.
+	 * @throws IllegalStateException If the EnumSelector has not been {@link #lock() locked}.
+	 */
+	public boolean allowsAll() {
+		checkReadable();
+		return defaultAllow && exceptions.isEmpty();
+	}
+
+	/**
+	 * Check if the EnumSelector blocks all values.
+	 *
+	 * @return If the EnumSelector blocks all values.
+	 * @throws IllegalStateException If the EnumSelector has not been {@link #lock() locked}.
+	 */
+	public boolean blocksAll() {
+		checkReadable();
+		return defaultBlock && exceptions.isEmpty();
+	}
+
+	/**
+	 * Returns an iterator of all the allowed elements in this EnumSelector.
+	 *
+	 * @return The iterator.
+	 * @throws IllegalStateException If the EnumSelector has not been {@link #lock() locked}.
+	 */
+	@Override
+	public Iterator<T> iterator() {
+		checkReadable();
+		return defaultBlock ? exceptions.iterator() : EnumSet.complementOf(exceptions).iterator();
```

Comment:
TODO: Make into an unmodifiable Iterator.
Reason: Iterators can otherwise be used to alter the contents of the underlying Iterable.